core/image/goost_image.cpp

```diff
@@ -504,6 +525,9 @@ bool GoostImage::has_pixelv(Ref<Image> p_image, const Vector2 &p_pos) {
 }
 
 bool GoostImage::get_pixel_or_null(Ref<Image> p_image, int x, int y, Color *r_pixel) {
+#ifdef DEBUG_ENABLED
+	ERR_FAIL_COND_V(p_image.is_null(), false);
+#endif
```

Comment:
Using checking in debug builds here, because it would be quite costly do check for valid image in release builds just for pixel access.